In [2]:
import re
import json
import pandas as pd

In [3]:
corpus=pd.read_csv("Macocu-sl-en-doc-format.csv", sep="\t", index_col = 0)
corpus

biroamer_entities translation_direction  \
2584979                No               sl-orig   
1212933                No               sl-orig   
598330                Yes               sl-orig   
374561                 No               sl-orig   
642105                Yes               sl-orig   
...                   ...                   ...   
3158106                No               sl-orig   
895254                 No               sl-orig   
2869258                No               sl-orig   
1740197                No               en-orig   
2414927                No               en-orig   

                                                 en_source en_var_doc  \
2584979                    http://15.liffe.si/?lang_chg=en          B   
1212933                    http://16.liffe.si/?lang_chg=en          B   
598330                     http://17.liffe.si/?lang_chg=en          B   
374561                 http://1proti1.mg-lj.si/en/concept/        UNK   
642105                     http://2006.fdf.si/?lang_chg=en          B   
...                                                    ...        ...   
3158106  https://zv-slo.si/en/responses-to-erroneous-or...          B   
895254   https://zvem.ezdrav.si/limesurvey/index.php/11...        UNK   
2869258  https://zwave.si/index.php?_route_=EverSpring/...          B   
1740197  https://zwave.si/index.php?_route_=Qubino/ZMNHDD1        UNK   
2414927         https://zwave.si/index.php?_route_=ZMNHVD1        UNK   

        en_var_dom                                          sl_source  \
2584979          B                    http://15.liffe.si/?lang_chg=sl   
1212933          B       http://16.liffe.si/index.php?menu_item=domov   
598330           B                    http://17.liffe.si/?lang_chg=sl   
374561         UNK                   http://1proti1.mg-lj.si/koncept/   
642105           B  http://2006.fdf.si/?menu_item=podatki&amp;menu...   
...            ...                                                ...   
3158106          B  https://zv-slo.si/odgovori-na-napacne-ali-zava...   
895254         UNK  https://zvem.ezdrav.si/limesurvey/index.php/11...   
2869258          A  https://zwave.si/index.php?_route_=EverSpring/...   
1740197          A     https://zwave.si/index.php?_route_=QUB-ZMNHBD1   
2414927          A     https://zwave.si/index.php?_route_=QUB-ZMNHBD1   

                en_domain         sl_domain  average_score  \
2584979       15.liffe.si       15.liffe.si       0.936808   
1212933       16.liffe.si       16.liffe.si       0.900000   
598330        17.liffe.si       17.liffe.si       0.957875   
374561   1proti1.mg-lj.si  1proti1.mg-lj.si       0.947393   
642105        2006.fdf.si       2006.fdf.si       0.941667   
...                   ...               ...            ...   
3158106         zv-slo.si         zv-slo.si       0.904375   
895254     zvem.ezdrav.si    zvem.ezdrav.si       0.958571   
2869258          zwave.si          zwave.si       0.910833   
1740197          zwave.si          zwave.si       0.860000   
2414927          zwave.si          zwave.si       0.837800   

                                                    en_doc  \
2584979  It went out with a bang. The evening sparkled ...   
1212933  Some days ago the organisers of the 17th Liffe...   
598330   17th LIFFe was brought to an end with the best...   
374561   Not interested in showcasing art, 1:1 would be...   
642105   The 8th International Documentary Film Festiva...   
...                                                    ...   
3158106  Responses to erroneous or misleading arguments...   
895254   Your sincere responses will help us adjust our...   
2869258  This presence detector using PIR (passive infr...   
1740197  But that's not all, since this module also has...   
2414927  Qubino Dimmer 0-10V But this is not all, since...   

                                                    sl_doc  en_length  \
2584979  Končalo se je razburljivo in z razkošjem. Veče...        574   
12129

In [4]:
# Calculate ratio of punctuations per words

def paragraph_punct_ratio(text):
    token_re=re.compile(r'\w+|\S',re.UNICODE)
    tokens=token_re.findall(text)
    punct=len([e for e in tokens if e in '.;,!?:'])
    ratio = punct/len(tokens)
    return ratio

corpus["punct_ratio"] = corpus.en_doc.apply(paragraph_punct_ratio)

corpus.head(3)

biroamer_entities translation_direction  \
2584979                No               sl-orig   
1212933                No               sl-orig   
598330                Yes               sl-orig   

                               en_source en_var_doc en_var_dom  \
2584979  http://15.liffe.si/?lang_chg=en          B          B   
1212933  http://16.liffe.si/?lang_chg=en          B          B   
598330   http://17.liffe.si/?lang_chg=en          B          B   

                                            sl_source    en_domain  \
2584979               http://15.liffe.si/?lang_chg=sl  15.liffe.si   
1212933  http://16.liffe.si/index.php?menu_item=domov  16.liffe.si   
598330                http://17.liffe.si/?lang_chg=sl  17.liffe.si   

           sl_domain  average_score  \
2584979  15.liffe.si       0.936808   
1212933  16.liffe.si       0.900000   
598330   17.liffe.si       0.957875   

                                                    en_doc  \
2584979  It went out with a bang. The evening sparkled ...   
1212933  Some days ago the organisers of the 17th Liffe...   
598330   17th LIFFe was brought to an end with the best...   

                                                    sl_doc  en_length  \
2584979  Končalo se je razburljivo in z razkošjem. Veče...        574   
1212933  Pred dnevi smo se iz 59. mednarodnega filmskeg...        293   
598330   S podelitvijo nagrad in predvajanjem Režiserja...        445   

         sl_length  punct_ratio  
2584979        463     0.103501  
1212933        184     0.076220  
598330         418     0.073930

In [6]:
corpus.punct_ratio.describe()

count    103281.000000
mean          0.094771
std           0.032344
min           0.000000
25%           0.075000
50%           0.089888
75%           0.107865
max           0.441536
Name: punct_ratio, dtype: float64

In [36]:
# With the ratio < 0.015 we catch non-textual texts without any punctuation
for i in corpus.query("punct_ratio < 0.015").en_doc.to_list()[:5]:
	print(i)

Council Directive 87/64/EEC of 30 December 1986 amending Directive 72/461/EEC on health problems affecting intra- Community trade in fresh meat and Directive 72/462/EEC on health and veterinary inspection problems upon importation of bovine animals and swine and fresh meat from third countries Council Directive 87/64/EEC of 30 December 1986 amending Directive 72/461/EEC on health problems affecting intra- Community trade in fresh meat and Directive 72/462/EEC on health and veterinary inspection problems upon importation of bovine animals and swine and fresh meat from third countries Direktiva Sveta z dne 21. Commission Directive 91/356/EEC of 13 June 1991 laying down the principles and guidelines of good manufacturing practice for medicinal products for human use
Decision No 1/89 of the EEC-EFTA Joint Committee 'Simplification of Formalities' of 3 May 1989 amending Annex II to the Convention on the simplification of formalities in trade in goods Decision No 2/89 of the EEC/EFTA Joint C

In [48]:
# With ratio above 0.2, we catch non-textual texts with a lot of punctuations
for i in corpus.query("punct_ratio > 0.20").en_doc.to_list()[:5]:
	print(i)

He is one of the most attractive and sought-after European percussionists, who has already recorded 6 of his own albums and has cooperated in more than 80 other albums. He has been cooperating with many renowned jazz artists, such as Dave Liebman, Louis Sclavis, Al di Meola, Michel Godard, Kenny Wheeler, Vincent Courtois, Ralph Alessi, Lee Konitz, Tim Berne, Ben Monder, Ravi Coltrane, Stefano Battaglia, Hal Crook, Mika Pohjola, George Garzone, Luciano Biondini, Norma Winstone, Drew Gress, Mick Goodrick, Erik Friedlander and many others.
The organizer of the festival reserves the right to change the program. Sunday, 27th June 2010, at 9.30 p.m. Bevkov Square, Nova Gorica Tomaž Nedoh &amp; band Project Bonaca www.nedoh.com Monday, 28th June 2010, at 8.30 p.m. Nova Gorica Music School Hall Maja Lisac, saxophone Gusets: Bojana Šaljiæ Podešva, live electronics Matjaž Drevenšek, saxophone Contemporary Saxophone Music Wednesday, 30th June 2010, at 9.00 p.m. Klinec Tourist Farm in Medana Lado 

In [49]:
# Filter the corpus by using only instances with ratio between 0.015 and 0.3
corpus = corpus.query("punct_ratio >= 0.015 & punct_ratio <= 0.2")

corpus.head()

biroamer_entities translation_direction  \
2584979                No               sl-orig   
1212933                No               sl-orig   
598330                Yes               sl-orig   
374561                 No               sl-orig   
642105                Yes               sl-orig   

                                   en_source en_var_doc en_var_dom  \
2584979      http://15.liffe.si/?lang_chg=en          B          B   
1212933      http://16.liffe.si/?lang_chg=en          B          B   
598330       http://17.liffe.si/?lang_chg=en          B          B   
374561   http://1proti1.mg-lj.si/en/concept/        UNK        UNK   
642105       http://2006.fdf.si/?lang_chg=en          B          B   

                                                 sl_source         en_domain  \
2584979                    http://15.liffe.si/?lang_chg=sl       15.liffe.si   
1212933       http://16.liffe.si/index.php?menu_item=domov       16.liffe.si   
598330                     http://17.liffe.si/?lang_chg=sl       17.liffe.si   
374561                    http://1proti1.mg-lj.si/koncept/  1proti1.mg-lj.si   
642105   http://2006.fdf.si/?menu_item=podatki&amp;menu...       2006.fdf.si   

                sl_domain  average_score  \
2584979       15.liffe.si       0.936808   
1212933       16.liffe.si       0.900000   
598330        17.liffe.si       0.957875   
374561   1proti1.mg-lj.si       0.947393   
642105        2006.fdf.si       0.941667   

                                                    en_doc  \
2584979  It went out with a bang. The evening sparkled ...   
1212933  Some days ago the organisers of the 17th Liffe...   
598330   17th LIFFe was brought to an end with the best...   
374561   Not interested in showcasing art, 1:1 would be...   
642105   The 8th International Documentary Film Festiva...   

                                                    sl_doc  en_length  \
2584979  Končalo se je razburljivo in z razkošjem. Veče...        574   
1212933  Pred dnevi smo se iz 59. mednarodnega filmskeg...        293   
598330   S podelitvijo nagrad in predvajanjem Režiserja...        445   
374561   Koncept 1:1 noče biti razstava v smislu izložb...        769   
642105   S slovenskim filmom Poročno potovanje se je ko...        423   

         sl_length  punct_ratio  
2584979        463     0.103501  
1212933        184     0.076220  
598330         418     0.073930  
374561         650     0.115178  
642105          13     0.084567

In [50]:
corpus.describe()

average_score      en_length      sl_length    punct_ratio
count  101807.000000  101807.000000  101807.000000  101807.000000
mean        0.897452     428.811084     495.158761       0.092997
std         0.063443    1694.062268    2320.090506       0.027555
min         0.502000      75.000000       2.000000       0.015000
25%         0.868429     119.000000      93.000000       0.074830
50%         0.913667     190.000000     165.000000       0.089552
75%         0.942684     346.000000     324.000000       0.106952
max         0.990500   98761.000000   94079.000000       0.200000

We remove 1474 texts and are left with 101,807 texts.

In [53]:
# Save the new file
corpus.to_csv("Macocu-sl-en-doc-format-filtered.csv", sep= "\t")

In [52]:
# Inspect whether this solved any issues with non-textual texts
non_textual_index = [150635, 2050301, 1738274, 2754062, 84675]

discarded_counter = 0

remaining_index = []

for i in non_textual_index:
	try:
		print(corpus.loc[i].en_doc)
		print(corpus.loc[i].punct_ratio)
		remaining_index.append(i)

	except KeyError:
		print(f"{i} not in final corpus.")
		discarded_counter += 1

print(discarded_counter)
print(remaining_index)

Dejan Habicht - Zvočni film / Sound Movie (00:05:22) Subtitle: 12 dolgočasnih pesmi Production: Center in Galerija P74, 2007 The sound of an airplane over a forest. Dejan Habicht - Nevarni vzorci Subtitle: 12 dolgočasnih pesmi Production: Center in Galerija P74, 2006 Dangerous Patterns is an electronic book, which can be printed at home (557KB). Dejan Habicht - die Heißeste Nummer Subtitle: 12 dolgočasnih pesmi Production: Center in Galerija P74, 2004/06 die Heißeste Nummer / The Hottest Number is an e-book that one can print at home. Saša Spačal - 7K: new life form (00:10:13) Production: Saša Spačal, Muzej sodobne umetnosti Metelkova - MSUM+, 2016 The video conceptually reflects on 7K: new life form, an artwork by Saša Spačal.
0.1518987341772152
2050301 not in final corpus.
1738274 not in final corpus.
Instructions for use of QR Code for FOTO LADO d.o.o. Click here for further information. Instructions for use of QR Code for STROPNIK RADO s.p. Instructions for use of QR Code for STROP

By filtering based on punctuation, we got rid of 2 out of 5 remaining non-textual texts.